### PR TITLE
podman network create: validate user input

### LINKED ISSUE #4184: podman network create: should validate input
```
# podman network create 'foo '   <--- foo with a space
/etc/cni/net.d/foo .conflist

# podman network create '../bar'
/etc/cni/bar.conflist
```
Present as of e67e9e12

### PR DESCRIPTION
**Pull request by Mrigank11**
Disallow invalid/confusing names such as '../bar' or 'foo '
Closes #4184

Signed-off-by: Mrigank Krishan <mrigankkrishan@gmail.com>

### CODE REVIEW

File: cmd/podman/network_create.go [hunk 1]
```diff
@@ -15,6 +16,9 @@ import (
 )
 
 var (
+	nameRegex  = regexp.MustCompile("^[a-zA-Z0-9][a-zA-Z0-9_.-]*$")
```
Comment (mheon) [inline cmd/podman/network_create.go]: Probably better to make the ones in Libpod public and use them here

File: cmd/podman/network_create.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"github.com/containers/libpod/pkg/network"
```
Comment (mheon) [inline cmd/podman/network_create.go]: While you're in here, mind moving this to the bottom with the other imports?

**Comment by TomSweeneyRedHat** [comment]
Other than @mheon's comments and the gating issue, LGTM

**Comment by baude** [comment]
what is wrong with a network named `foo` ?

**Comment by mheon** [comment]
@baude Trailing space

**Comment by mheon** [comment]
LGTM

**Pull request merged**